### PR TITLE
Allow conditionally excluding views from the DashContainer

### PR DESCRIPTION
**Pull request by TomTirapani**
+ Allow preventing adding views to the DashContainer

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/dash/DashViewSpec.js [hunk 1]
```diff
@@ -31,7 +32,11 @@ export class DashViewSpec {
      * @param {string} title - Title text added to the tab header.
      * @param {Icon} [icon] - An icon placed at the left-side of the tab header.
      * @param {boolean} [unique] - true to prevent multiple instances of this view. Default false.
-     * @param {boolean} [allowClose] - true (default) to allow removing from the DashContainer.
+     * @param {boolean} [exclude] - true to prevent any instances of this view. References to this
+     *      view in state will be quietly dropped. Default false.
+     * @param {boolean} [allowAdd] - true (default) to allow adding new instances of this view.
+     *      References to this view in state will be respected.
+     * @param {boolean} [allowClose] - true (default) to allow removing instances from the DashContainer.
```
Comment (lbwexler) [inline desktop/cmp/dash/DashViewSpec.js]: allowClose => allowRemove for symmetry with allowAdd?

File: desktop/cmp/dash/DashViewSpec.js [hunk 2]
```diff
@@ -31,7 +32,11 @@ export class DashViewSpec {
...
      * @param {RenderMode} [renderMode] - strategy for rendering this DashTab. If null, will
```
Comment (lbwexler) [inline desktop/cmp/dash/DashViewSpec.js]: remove stale references to DashTab

**Comment by TomTirapani** [comment]
Both great suggestions - both done

**Comment by lbwexler** [comment]
Regarding the 'exclude' part that is the core of this -- why not have the app just drop the viewSpec before passing to the co

We could also use the term `omit` and just drop them right in the constructor, if we wanted to make things cleaner for the app, but still keep things simple.

**Pull request merged**